### ISSUE TITLE
MD051 false positive `## $\LaTeX$ hooks` ⟼ `[latex hooks](#latex-hooks)`

### ISSUE THREAD

**Issue by randolf-scholz** [DavidAnson/markdownlint#1152]
Example markdown

```markdown
# custom pre-commit hooks

[latex hooks](#latex-hooks)

## $\LaTeX$ Hooks
```

Linking works without a problem.

**Comment by DavidAnson** [comment]
The use of math syntax in the heading text seems to be interfering. I'll have a look soon.

https://dlaa.me/markdownlint/#%25m%23%20custom%20pre-commit%20hooks%0A%0A%5Blatex%20hooks%5D(%23latex-hooks)%0A%0A%23%23%20%24%5CLaTeX%24%20Hooks%0A
